packages/block-editor/src/components/global-styles/dimensions-panel.js

```diff
@@ -0,0 +1,520 @@
+	const hasPadding = useHasPadding( settings );
+	const hasMargin = useHasMargin( settings );
+	const hasGap = useHasGap( settings );
+	const hasMinHeight = useHasMinHeight( settings );
+
+	return (
+		hasContentSize ||
+		hasWideSize ||
+		hasPadding ||
+		hasMargin ||
+		hasGap ||
+		hasMinHeight
+	);
+}
+
+function useHasContentSize( settings ) {
+	return settings?.layout?.contentSize;
+}
+
+function useHasWideSize( settings ) {
+	return settings?.layout?.wideSize;
+}
+
+function useHasPadding( settings ) {
+	return settings?.spacing?.padding;
+}
+
+function useHasMargin( settings ) {
+	return settings?.spacing?.margin;
+}
+
+function useHasGap( settings ) {
+	return settings?.spacing?.blockGap;
+}
+
+function useHasMinHeight( settings ) {
+	return settings?.dimensions?.minHeight;
+}
+
+function useHasSpacingPresets( settings ) {
+	const {
+		custom,
+		theme,
+		default: defaultPresets,
+	} = settings?.spacing?.spacingSizes || {};
+	const presets = custom ?? theme ?? defaultPresets ?? [];
+
+	return presets.length > 0;
+}
+
+function filterValuesBySides( values, sides ) {
+	if ( ! sides ) {
+		// If no custom side configuration all sides are opted into by default.
+		return values;
+	}
+
+	// Only include sides opted into within filtered values.
+	const filteredValues = {};
+	sides.forEach( ( side ) => {
+		if ( side === 'vertical' ) {
+			filteredValues.top = values.top;
+			filteredValues.bottom = values.bottom;
+		}
+		if ( side === 'horizontal' ) {
+			filteredValues.left = values.left;
+			filteredValues.right = values.right;
+		}
+		filteredValues[ side ] = values[ side ];
+	} );
+
+	return filteredValues;
+}
+
+function splitStyleValue( value ) {
+	// Check for shorthand value (a string value).
+	if ( value && typeof value === 'string' ) {
+		// Convert to value for individual sides for BoxControl.
+		return {
+			top: value,
+			right: value,
+			bottom: value,
+			left: value,
+		};
+	}
+
+	return value;
+}
+
+function splitGapValue( value ) {
+	// Check for shorthand value (a string value).
+	if ( value && typeof value === 'string' ) {
+		// If the value is a string, treat it as a single side (top) for the spacing controls.
+		return {
+			top: value,
+		};
+	}
+
+	if ( value ) {
+		return {
+			...value,
+			right: value?.left,
+			bottom: value?.top,
+		};
+	}
+
+	return value;
+}
+
+function DimensionsToolsPanel( props ) {
+	return <ToolsPanel label={ __( 'Dimensions' ) } { ...props } />;
+}
+
+const DEFAULT_CONTROLS = {
+	contentSize: true,
+	wideSize: true,
+	padding: true,
+	margin: true,
+	blockGap: true,
+	minHeight: true,
+};
+
+export default function DimensionsPanel( {
+	as: Wrapper = DimensionsToolsPanel,
+	value,
+	onChange,
+	inheritedValue = value,
+	settings,
+	panelId,
+	defaultControls = DEFAULT_CONTROLS,
+} ) {
+	const decodeValue = ( rawValue ) =>
+		getValueFromVariable( { settings }, '', rawValue );
+
+	const showSpacingPresetsControl = useHasSpacingPresets( settings );
```

Comment:
It appears that `useHasSpacingPresets` is only returning true for displaying the spacing presets control when in the Global Styles, not when styling individual blocks.

Reply:
Should be fixed, good catch